torch/autograd/__init__.py

```diff
@@ -26,32 +26,50 @@
 __all__ = ['Variable', 'Function', 'backward', 'grad_mode']
 
 _OptionalTensor = Optional[torch.Tensor]
+_ShapeorNestedShape = Union[_size, Sequence[_size]]
+
+
+def _calculate_shape(output: torch.Tensor, grad: torch.Tensor,
+                     is_grads_batched: bool) -> Tuple[_ShapeorNestedShape, _ShapeorNestedShape]:
+    if output.is_nested:
+        if is_grads_batched:
+            raise RuntimeError("Batched grads are not supported with Nested Tensor.")
+
+        out_shape = [tensor.size() for tensor in output.unbind()]
```

Comment:
I think @albanD left a comment earlier on this. What was the resolution on that? That this is okay as long as nested tensors aren't twice nested?

Reply:
I don't think this is ok. We should have a proper way to print this.
If nested has this assumption today, that is fine. But that should be in nested-specific code and not in user code.

Reply:
Yeah this should always work as long as nested tensors aren't twice nested and core doesn't currently support twice nested tensors

Reply:
#79553
Split of the changes around this area of the autograd into separate PR